backend/groth16/bls12-377/prove.go

```diff
@@ -102,6 +101,15 @@ func Prove(r1cs *cs.R1CS, pk *ProvingKey, fullWitness witness.Witness, opts ...b
 
 	start := time.Now()
 
+	commitmentsSerialized := make([]byte, fr.Bytes*len(r1cs.CommitmentInfo))
```

Comment:
if we commit to many variables, that could slow the prover. It's also running in the "main go routine" of the prover. Any way to avoid the `[]byte` allocation?

Reply:
This is not too bad imo. Even though the number of committed variables may be huge, so far the number of _commitments_ themselves have been pretty small. I suggest we keep an eye on it until we find a use-case where there is at least $O(\log n)$ or so commitments.